packages/nimble-components/src/table/models/table-layout-manager.ts

```diff
@@ -0,0 +1,263 @@
+        }
+    }
+
+    private performCascadeSizeRight(
+        rightColumnIndex: number,
+        delta: number
+    ): void {
+        let currentDelta = delta;
+        const rightColumnInitialWidths = this.initialColumnPixelWidths[rightColumnIndex]!;
+        const allowedDelta = delta > 0
+            ? Math.min(
+                rightColumnInitialWidths.initialPixelWidth
+                          - rightColumnInitialWidths.minPixelWidth,
+                currentDelta
+            )
+            : delta;
+        const actualDelta = Math.round(allowedDelta);
+        const visibleColumns = this.getVisibleColumns();
+        const rightColumn = visibleColumns[rightColumnIndex]!;
+        rightColumn.columnInternals.currentPixelWidth! -= actualDelta;
+
+        if (
+            actualDelta < currentDelta
+            && rightColumnIndex < visibleColumns.length - 1
+            && delta > 0
+        ) {
+            currentDelta -= allowedDelta;
+            this.performCascadeSizeRight(rightColumnIndex + 1, currentDelta);
+        }
+    }
+
+    private flagActiveColumnDividers(): void {
+        const visibleColumnLength = this.getVisibleColumns().length;
+        const leftDividerIndex = this.activeColumnDivider! > 0
+            ? this.activeColumnDivider!
+                  - ((this.activeColumnDivider! + 1) % 2)
+            : 0;
+        const rightDividerIndex = leftDividerIndex === 0
+            || leftDividerIndex === visibleColumnLength + 1
+            ? leftDividerIndex
+            : leftDividerIndex + 1;
+        const dividers = this.table.shadowRoot!.querySelectorAll('.column-divider');
+        Array.from(dividers).forEach((divider, i) => {
+            if (i === leftDividerIndex || i === rightDividerIndex) {
+                divider.setAttribute('active', 'true');
+            }
+        });
+    }
+
+    private unflagActiveColumnDividers(): void {
+        const dividers = this.table.shadowRoot!.querySelectorAll('.column-divider');
+        Array.from(dividers).forEach(divider => {
+            divider.removeAttribute('active');
+        });
+    }
+
+    private cacheGridSizedColumns(): void {
+        this.gridSizedColumns = [];
+        for (const column of this.getVisibleColumns()) {
+            if (column.columnInternals.currentPixelWidth === undefined) {
+                this.gridSizedColumns.push(column);
+            }
+        }
+    }
+
+    private cacheColumnInitialPixelWidths(): void {
+        for (const column of this.getVisibleColumns()) {
+            this.initialColumnPixelWidths.push({
+                initialPixelWidth: column.columnInternals.currentPixelWidth!,
+                minPixelWidth: column.columnInternals.minPixelWidth
+            });
+        }
+    }
+
+    private resetGridSizedColumns(): void {
+        if (!this.gridSizedColumns) {
+            return;
+        }
+
+        const largestColumnFixedSize = Math.max(
```

Comment:
The algorithm has changed to simply be:
currentFractionalWidth = (currentPixelWidth / initialPixelWidth) * initialFractionalWidth

So, now, if we have a series of fractional width columns: [1, 1, 1, 1], with an initial table total column width of 400 (and no minimum column sizes), and then size the first column by 150 pixels (it started off as 100), then the resulting fractional widths will be [2.5, .5, .5, .5].

This means that if I add a new column after the above sizing operation, it will have its expected default render width when considering its default fractional width of 1 (be twice as big as the smallest columns).